server/src/main/java/io/crate/expression/symbol/LiteralValueFormatter.java

```diff
@@ -50,8 +50,11 @@ public void formatValue(Object value, StringBuilder builder) {
             formatIterable((Iterable<?>) value, builder);
         } else if (value.getClass().isArray()) {
             formatArray(value, builder);
-        } else if (value instanceof String || value instanceof Point || value instanceof Period) {
+        } else if (value instanceof String || value instanceof Point) {
             builder.append(Literals.quoteStringLiteral(value.toString()));
+        } else if (value instanceof Period) {
+            builder.append(Literals.quoteStringLiteral(value.toString()));
```

Comment:
That could probably also be handled by a conversion back from `Period` to `IntervalLiteral` which would save one implicit cast.

Reply:
However, internal representation in CrateDB is a `Period` for an Interval, so this solution could be also acceptable.